core/src/main/java/org/frankframework/util/MessageUtils.java

```diff
@@ -177,7 +163,7 @@ public static Charset computeDecodingCharset(Message message, int confidence) th
 		}
 
 		CharsetDetector detector = new CharsetDetector();
-		detector.setText(message.getMagic());
+		detector.setText(message.asInputStream());
```

Comment:
PS: The API doc says that this also does a mark-and-reset of the input stream so this is safe, just like Tika with inputstream.